src/tests/fs_tests.rs

```diff
@@ -889,4 +891,34 @@ mod fs_tests {
         assert_eq!(cage.exit_syscall(EXIT_SUCCESS), EXIT_SUCCESS);
         lindrustfinalize();
     }
+
+    pub fn ut_lind_fs_dir_chdir_getcwd() {
+        lindrustinit();
+        let cage = {CAGE_TABLE.read().unwrap().get(&1).unwrap().clone()};
+        let needed = "/subdir1\0".as_bytes().to_vec().len();
+
+        use std::convert::TryInto;
+        let needed_u32: u32 = needed.try_into().unwrap();
+
+        let mut buf = vec![0u8; needed];
+        let bufptr: *mut u8 = &mut buf[0];
+
+        assert_eq!(cage.chdir_syscall("/"), 0);
+        assert_eq!(cage.getcwd_syscall(bufptr, 0), -(Errno::ERANGE as i32));
+        assert_eq!(cage.getcwd_syscall(bufptr, 1), -(Errno::ERANGE as i32));
+        assert_eq!(cage.getcwd_syscall(bufptr, 2), 0);
+        assert_eq!(String::from_utf8_lossy(&buf), "/\0\0\0\0\0\0\0\0");
+
+        cage.mkdir_syscall("/subdir1", S_IRWXA);
+        assert_eq!(cage.access_syscall("subdir1", F_OK), 0);
+        assert_eq!(cage.chdir_syscall("subdir1"), 0);
+
+        assert_eq!(cage.getcwd_syscall(bufptr, 0), -(Errno::ERANGE as i32));
+        assert_eq!(cage.getcwd_syscall(bufptr, needed_u32-1), -(Errno::ERANGE as i32));
+        assert_eq!(cage.getcwd_syscall(bufptr, needed_u32), 0);
+        assert_eq!(String::from_utf8_lossy(&buf), "/subdir1\0");
```

Comment:
You can/should probably use the non lossy versions of these.

Reply:
Why do we not want the lossy versions? Is a panic preferable to an un-exact string?

Reply:
In the tests we know that you're passing/getting passed valid utf8, so it won't return an error variant (which I assume is what you mean by panic as that would panic on an unwrap). Additionally, the lossy and non-lossy versions actually do somewhat different things--the non-lossy version allocates a new buffer whereas the lossy version does it in place and wraps it within a Cow<> object which makes it copy-on-write. This means it's somewhat more complex and annoying to deal with, and we don't care about the overhead of the utf8 validity check within the test.

Reply:
Updated this to use std::str::from_utf8.

(String::from_utf8 does not work directly as it requires the ownership of the passed vector.)